### PR TITLE
Small bug fixes + minor code changes

### PR DESCRIPTION
**Pull request by 0ct0ber**
These are mostly small bugfixes, where code isn't running how it seems to be intended, or wrong classtypes were used on checks (for example: checking if a list of TempBlocks contains a block), and some NullPointerException issues. Also, PresetCommand wasn't properly checking if the player has met their maximum presets yet (which is a significant bug).

Also, I've found it's really helpful if you make changes in your IDE first, as you will be able to see where the wrong classtype has been used incorrectly in collections when using contains(object).

##Fixes

Fixed PresetCommand not properly checking if the player has met their maximum presets (this check would never be raised), due to player being used instead of the player's UUID in a check
Fixed potions possibly being drained on rare occasions in WaterReturn#emptyWaterBottle, now even rarer
Fixed NullPointerExceptions in WaterReturn#emptyWaterBottle that occur almost every single time, and optimized code
Fixed WaterSpout reverting the base block briefly where this isn't intended, due to an improper check (Block instead of TempBlock)
##Changes

Changed WaterReturn#emptyWaterBottle from void to boolean (returns whether or not the bottle was successfully removed), so that other classes or abilities can check if it wasn't removed where expected

### CODE REVIEW

File: src/com/projectkorra/projectkorra/waterbending/util/WaterReturn.java [hunk 1]
```diff
@@ -148,42 +148,38 @@ public static boolean hasWaterBottle(Player player) {
 		if (inventory.contains(Material.POTION)) {
 			ItemStack item = inventory.getItem(inventory.first(Material.POTION));
 			PotionMeta meta = (PotionMeta) item.getItemMeta();
-			return meta.getBasePotionData().getType().equals(PotionType.WATER);
+			return meta.getBasePotionData().getType() == PotionType.WATER);
```
Comment (ChristopherWMM) [inline src/com/projectkorra/projectkorra/waterbending/util/WaterReturn.java]: It appears as though you have an extra/missing parenthesis here.

Comment (0ct0ber) [reply src/com/projectkorra/projectkorra/waterbending/util/WaterReturn.java]: Oh, thanks!

**Comment by 0ct0ber** [comment]
@ChristopherWMM are you going to add either of these pull requests?

**Comment by ChristopherWMM** [comment]
No, most of these are patches at best. I have gone in and resolved a lot of the underlying issues myself using these PR's as a reference. Many of these issues will be resolved in my next PR.

**Comment by 0ct0ber** [comment]
@ChristopherWMM I'm just trying to help... a month ago I was wanting to fix issues and make as little code changes as necessary, and then make more significant code changes or new mechanics afterwards, in separate pull requests from just the fixes.

It's a bit discouraging when you're so particular with the PRs; you closed one just because I didn't optimize Illumination and fix it significantly enough (but remember, I'm still just trying to help- and my pull request was of help, it just wasn't as useful as you would have done it yourself). I did optimize it in my other pull request still open, and I think you're not accepting these because you have some sort of bias towards me; you've also obviously been addressing other pull requests and issues since the month ago I opened these.

No I'm not a noob at this; I've been coding plugins for years, since 2015, and just because I make a few typos by not using an editor for such small changes doesn't mean I'm not careful. If you look at a lot of the code right now, there are mistakes that occurred because the people writing it didn't remember to check for NPEs, or didn't even bother to put their code in an editor (yes no compile errors occurred, but I found a number of wrong collection checks by just cloning into Eclipse). I'm fixing errors that occurred due to bad code, and am being precise in the process to eliminate the errors without causing any side effects or requiring the whole branch to be changed.

**Comment by ChristopherWMM** [comment]
I think you misunderstood my last comment. I am not trying to attack you personally when I leave comments on your code. These have been very helpful at pointing out issues as well as where they are potentially occurring. I am not being biased towards you or your contributions, it is just that these fixes either do not completely solve the underlying issues or leave extraneous/unfinished code in the branch. I would recommend for future PR's you split up each fix into its own separate request instead of mashing them all together, it makes it much easier for us to have a conversation and work with one another. If I find one thing in a huge PR it prevents me from accepting everything else.

**Comment by 0ct0ber** [comment]
@ChristopherWMM Alright... I just thought I was correcting what you had requested, and then you still weren't accepting the pull request (and didn't know why you hadn't commented for a month). I had wondered if it was because of some typos earlier.

I'm glad you've gone and fixed these bugs anyways.

**Pull request closed without merging**